From 2de8b12e3f25eb9717a240fde8edb4e2ac0cb0a0 Mon Sep 17 00:00:00 2001
From: Daniel Mueller <deso@posteo.net>
Date: Sun, 24 Sep 2017 15:01:38 -0700
Subject: [PATCH 01/16] Do not wrap around when focusing another container

When focusing the next container to the right we may actually focus one
that is left of the current one if we already reached the right most
container.
This behavior is undesired. This patch adjusts the focusing logic to no
longer wrap around. If we reached the right most container and issue a
'focus right' command the focus will simply stay where it is. The
behavior when changing the focus in another direction (left/up/down) is
adjusted in a similar way.
---
 src/tree.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tree.c b/src/tree.c
index d5f457..cb30bd 100644
--- a/src/tree.c
+++ b/src/tree.c
@@ -675,7 +675,7 @@ static bool _tree_next(Con *con, char way, orientation_t orientation, bool wrap)
  *
  */
 void tree_next(char way, orientation_t orientation) {
-    _tree_next(focused, way, orientation, true);
+    _tree_next(focused, way, orientation, false);
 }
 
 /*
-- 
2.16.1

